src/components/CodeRevokeModal/index.jsx

```diff
@@ -138,130 +144,59 @@ class CodeRevokeModal extends React.Component {
+    const { doNotEmail, mode } = this.state;
 
     return (
       <>
-        {submitFailed && this.renderErrorMessage()}
+        {submitFailed
+          && (
+          <ModalError
+            title={ERROR_MESSAGE_TITLES[mode]}
+            errors={error}
+            ref={this.errorMessageRef}
+          />
+          )}
         <div className="assignment-details mb-4">
           {isBulkRevoke && (
             <>
-              {data.selectedCodes.length > 0 && <p className="bulk-selected-codes">Selected Codes: {data.selectedCodes.length}</p>}
+              {data.selectedCodes.length > 0 && <p className="bulk-selected-codes">{displaySelectedCodes(data.selectedCodes.length)}</p>}
             </>
           )}
           {!isBulkRevoke && this.hasIndividualRevokeData() && (
             <>
-              <p className="code">Code: {data.code}</p>
-              <p className="email">Email: {data.assigned_to}</p>
+              <p className="code">{displayCode(data.code)}</p>
+              <p className="email">{displayEmail(data.assigned_to)}</p>
             </>
           )}
         </div>
-        <form onSubmit={e => e.preventDefault()}>
-          <div className="mt-4">
-            <h3>Email Template</h3>
-            <TemplateSourceFields emailTemplateType="revoke" disabled={doNotEmail} />
-            <Field
-              id="email-template-subject"
-              name="email-template-subject"
-              component={RenderField}
-              type="text"
-              label="Customize Email Subject"
-              disabled={doNotEmail}
-            />
-            <Field
-              id="email-template-greeting"
-              name="email-template-greeting"
-              component={TextAreaAutoSize}
-              label="Customize Greeting"
-              disabled={doNotEmail}
-            />
-            <Field
-              id="email-template-body"
-              name="email-template-body"
-              component={TextAreaAutoSize}
-              label="Body"
-              disabled
-            />
-            <Field
-              id="email-template-closing"
-              name="email-template-closing"
-              component={TextAreaAutoSize}
-              label="Customize Closing"
-              disabled={doNotEmail}
-            />
-            <Field
-              name="do-not-email"
-              component={this.doNotEmailField}
-              onChange={(event, newValue) => {
-                this.setDoNotEmail(newValue);
-              }}
-            />
-
-          </div>
-        </form>
+        <EmailTemplateForm emailTemplateType={MODAL_TYPES.revoke} disabled={doNotEmail}>
+          <Field
+            name="do-not-email"
+            className="do-not-email-wrapper"
+            component={CheckboxWithTooltip}
+            label="Do not email"
+            checked={doNotEmail}
+            id="doNotEmailCheckbox"
+            icon={faInfoCircle}
+            altText="More information"
```

Comment:
is this alt text value really for the icon? Ie should we have alt text that specifies this field is about not emailing?

Reply:
Yeah it's just for the icon, the actual information is in the tooltip.